File: student-portal-frontend/src/services/TimetableService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/api/timetable";

export const getAllSchedules = () => {
  return axios.get(BASE_URL);
};

export const addSchedule = (schedule) => {
  return axios.post(BASE_URL, schedule);
};

export const updateSchedule = (id, schedule) => {
  return axios.put(`${BASE_URL}/${id}`, schedule);
};

export const deleteSchedule = (id) => {
  return axios.delete(`${BASE_URL}/${id}`);
};

export const getSchedulesByDay = (dayOfWeek) => {
  return axios.get(`${BASE_URL}/day/${dayOfWeek}`);
};

export const getCourseSchedule = (courseId) => {
  return axios.get(`${BASE_URL}/course/${courseId}`);
};

File: student-portal-frontend/StudentPortal/src/main/java/service/TimetableService.java
package service;

import org.springframework.stereotype.Service;

import model.Timetable;
import repository.TimetableRepository;

import java.util.List;

@Service
public class TimetableService {

    private final TimetableRepository repository;

    public TimetableService(TimetableRepository repository) {
        this.repository = repository;
    }

    public List<Timetable> getAllSchedules() {
        return repository.findAll();
    }

    public Timetable addSchedule(Timetable timetable) {
        return repository.save(timetable);
    }

    public Timetable updateSchedule(Timetable timetable) {
        return repository.save(timetable);
    }

    public void deleteSchedule(Long id) {
        repository.deleteById(id);
    }

    public List<Timetable> getSchedulesByDay(String dayOfWeek) {
        return repository.findByDayOfWeek(dayOfWeek);
    }

    public List<Timetable> getCourseSchedule(Long courseId) {
        return repository.findByCourseId(courseId);
    }
}